coalton-asdf.lisp

```diff
@@ -0,0 +1,28 @@
+;; ASDF Coalton component type
+;;
+;; Add an asdf component that supports compiling files with a .coalton
+;; extension.
+;;
+;; To use, add a :defsystem-depends-on clause to your system
+;; definition, and refer to files using component type :coalton-file,
+;; like:
+;;
+;;     (asdf:defsystem #:application
+;;       :defsystem-depends-on (#:coalton-asdf)
+;;       :depends-on (#:coalton)
+;;       :pathname "src/"
+;;       :serial t
+;;       :components ((:coalton-file "types")
+;;                    (:coalton-file "main")))
+;;
+;; This class is defined in the ASDF package, so that the keyword :coalton-file works.
+;;
+;; see: https://github.com/fare/asdf/blob/master/doc/best_practices.md#using-asdf-extensions
+
+(in-package :asdf)
+
+(defclass coalton-file (cl-source-file)
```

Comment:
is it kosher to define new things in ASDF? I think we should make a new package.

Reply:
asdf documentation suggests that this is a common practice, in: https://github.com/fare/asdf/blob/master/doc/best_practices.md#using-asdf-extensions 

> If you need an ASDF extension, the recommended way is to use :defsystem-depends-on. The extension will define new classes of operations or components, new functions, etc. If it defines them in the ASDF package, you can refer to them using a keyword of the same name, e.g. :cffi-wrapper-file for one of the component classes defined by system cffi-grovel, :static-program-op for one of the operation classes defined by cffi-toolchain, or :f2cl-system for the system class defined by system f2cl.

Reply:
absolutely goofy but that's what they recommend so that's what they recommend